### PR TITLE
macOS: Adapt SCI Certificate Tests Names and ResultData to match Windows

### PR DESCRIPTION
**Pull request by vl-federico**
See the updated [test protocol](https://podio.com/uberagentcom/uberagent-dev/apps/test-protocols/items/591) for sample output.

### CODE REVIEW

File: config-dev/Security inventory/macOS/Certificates/Certificates.zsh [hunk 1]
```diff
@@ -253,10 +262,47 @@ vlCheckKeychains()
...
+      local certObj="{}"
+      if [[ -n "$username" ]]; then
+        certObj=$( vlAddResultValue "${certObj}" "Scope" "User" )
+      else
+        certObj=$( vlAddResultValue "${certObj}" "Scope" "Machine" )
+      fi
+
+      certObj=$( vlAddResultValue "${certObj}" "SecurityScore" 5 )
+      certObj=$( vlAddResultValue "${certObj}" "RiskScore" $riskScore )
+      certObj=$( vlAddResultValue "${certObj}" "Issuer" "$issuer" )
```
Comment (martin-kretzschmar) [inline config-dev/Security inventory/macOS/Certificates/Certificates.zsh:297]: Can we alter the formatting to, e.g., `CN=some-cn, DC=some-dc, DC=another-dc, DC=final-dc`?

File: config-dev/Security inventory/macOS/Certificates/Certificates.zsh [hunk 2]
```diff
@@ -36,18 +36,22 @@ vlReportTrustStoreVersion()
 
 vlCheckSystemKeychainCerts()
 {
-  local testName="CertSysKeyChk"
-  local testDisplayName="System Keychain Cert Check"
+  local testName="CMTrByMac"
+  local testDisplayName="Certificates trusted by macOS - Machine"
```
Comment (martin-kretzschmar) [inline config-dev/Security inventory/macOS/Certificates/Certificates.zsh:40]: Overall test category name needs to be `Certificate`.

File: config-dev/Security inventory/macOS/Certificates/Certificates.zsh [hunk 3]
```diff
@@ -253,10 +262,47 @@ vlCheckKeychains()
...
+                      openssl x509 -noout -subject 2>/dev/null |
+                      sed 's/subject=//')
+
+      local thumbprint=$(echo "$singleCertData" |
+                        openssl x509 -noout -sha1 -fingerprint 2>/dev/null |
+                        tr -d ':' |
+                        cut -d= -f2)
+
+      local certObj="{}"
+      if [[ -n "$username" ]]; then
```
Comment (martin-kretzschmar) [inline config-dev/Security inventory/macOS/Certificates/Certificates.zsh:289]: Scope determination needs a double-check. Always returns `Machine` during my tests.

**Review by martin-kretzschmar — CHANGES_REQUESTED** [review]

File: config-dev/Security inventory/macOS/Certificates/Certificates.zsh (hunk 2 above)
Comment (vl-federico) [reply config-dev/Security inventory/macOS/Certificates/Certificates.zsh:40]: Sorry, I don't quite understand what you mean. Do you mean to modify the testName to "CertificateTrByMac"?

Comment (vl-federico) [reply config-dev/Security inventory/macOS/Certificates/Certificates.zsh:40]: Fixed in d232bd194a3d71b5c20b8e1a3d98724f0672a0b2.

File: config-dev/Security inventory/macOS/Certificates/Certificates.zsh (hunk 3 above)
Comment (vl-federico) [reply config-dev/Security inventory/macOS/Certificates/Certificates.zsh]: Fixed in 30bb39b6311c12ded8e7cf72aedaf0a37c084ca4. Also ensured that an empty certificate list results in the username being sent as `ResultData`.

File: config-dev/Security inventory/macOS/Certificates/Certificates.zsh (hunk 1 above)
Comment (vl-federico) [reply config-dev/Security inventory/macOS/Certificates/Certificates.zsh:339]: Using the root CA for vastlimits, I created a fix in 556bf069c11227dcc310c8810b679db0b6e2b49d.

**Comment by vl-federico** [comment]
@martin-kretzschmar Removed the `Scope` attribute, as discussed with Helge today in the [R&D meeting](https://podio.com/uberagentcom/intranet/apps/meetings/items/1144).

**Review by martin-kretzschmar — APPROVED** [review]

**Pull request merged**